build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/CreateFrameworkList.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class CreateFrameworkList : Task
+	{
+		[Required]
+		public                  ITaskItem       FrameworkListFile           { get; set; }
+
+		[Required]
+		public                  ITaskItem       FrameworkDirectory          { get; set; }
+
+		[Required]
+		public                  ITaskItem[]     FrameworkFiles              { get; set; }
+
+		public                  ITaskItem[]     FrameworkFileOverrides      { get; set; }
+
+		[Required]
+		public                  string          Redist                      { get; set; }
+
+		[Required]
+		public                  string          Name                        { get; set; }
+
+		public override bool Execute()
+		{
+			var files = new List<ITaskItem>(FrameworkFiles);
+			files.Sort ((x, y) => {
+				var a = Path.GetFileNameWithoutExtension (x.ItemSpec);
+				var b = Path.GetFileNameWithoutExtension (y.ItemSpec);
+				return string.Compare (a, b, StringComparison.OrdinalIgnoreCase);
+			});
+			var contents = new XElement ("FileList",
+					new XAttribute ("Redist", Redist),
+					new XAttribute ("Name", Name),
+					files.Select (f => ToFileElement (f)));
+			contents.Save (FrameworkListFile.ItemSpec);
+			return true;
+		}
+
+		XElement ToFileElement (ITaskItem file)
+		{
+			var path = Path.Combine (FrameworkDirectory.ItemSpec, file.ItemSpec);
+			if (!File.Exists (path)) {
+				path = Path.Combine (FrameworkDirectory.ItemSpec, "Facades", file.ItemSpec);
+			}
+			using (var pe = new PEReader (File.OpenRead (path))) {
+				var reader          = pe.GetMetadataReader();
+				var assemblyDef     = reader.GetAssemblyDefinition();
+
+				var taskVersion     = Nullable (file.GetMetadata ("Version"));
+				var overrideVersion = Nullable (FrameworkFileOverrides.FirstOrDefault (o => o.ItemSpec == file.ItemSpec)?.GetMetadata ("Version"));
+				var assemblyVersion = assemblyDef.Version.ToString ().TrimStart ('v');
```

Comment:
`assemblyDef.Version` is a `System.Version`, I don't think you need the `TrimStart()`?

Reply:
We need a string, hence the `.ToString()` in there, but `Version.ToString()` inserts a leading `v`, which we don't want in the XML.

Reply:
I've not seen it do that? `Console.WriteLine(new Version ("1.0.0").ToString());` just prints `1.0.0` for me.

Reply:
I think I'm confusing it with a different type then. :-(